generator/src/main/java/com/scottlogic/deg/generator/generation/StandardFieldValueSourceEvaluator.java

```diff
@@ -108,20 +108,42 @@ private FieldValueSource getNumericSource(FieldSpec fieldSpec) {
             ? new NumericRestrictions()
             : fieldSpec.getNumericRestrictions();
 
-        int numericScale = fieldSpec.getGranularityRestrictions() != null
-            ? fieldSpec.getGranularityRestrictions().getNumericScale()
-            : 0;
+        int numericScale = getNumericScale(restrictions, fieldSpec);
 
-        if (numericScale == 0) {
+        if ((restrictions.min == null || restrictions.max == null) || isIntegerFieldValue(restrictions, numericScale)) {
```

Comment:
What does this do if restrictions.min is null, 
but restrictions.max is a decimal?

Why is the existence of max values related to it being an integer?

Reply:
I'll add some extra tests to cover this but this essentially keeps the existing functionality where a range hasn't been created which gets used when any value can be generated. Think scenarios where strings, temporal and numeric values are generated.